esp/services/ws_access/ws_accessService.cpp

```diff
@@ -37,11 +37,14 @@
 #define MAX_RESOURCES_DISPLAY 3000
 static const long MAXXLSTRANSFER = 5000000;
 
-void checkUser(IEspContext& context, const char* rtype = NULL, const char* rtitle = NULL, unsigned int SecAccessFlags = SecAccess_Full)
+void Cws_accessEx::checkUser(IEspContext& context, CLdapSecManager* secmgr, const char* rtype, const char* rtitle, unsigned int SecAccessFlags)
 {
-    CLdapSecManager* secmgr = dynamic_cast<CLdapSecManager*>(context.querySecManager());
-    if(secmgr == NULL)
-        throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+    if (secmgr == nullptr)
+    {
+        secmgr = queryLDAPSecurityManager(context);
+        if (secmgr == nullptr)
+            throw makeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
```

Comment:
If you add a flag to queryLDAPSecurityManager(IEspContext &context, bool required) which tells it to throw the exception if it results in nullptr then you can probably eliminate all of these duplicate throw lines.

Could be another JIRA though.

Reply:
I see you got rid of most of them in this code already using queryLDAPSecurityManagerAndCheckUser()... you can decide if moving it up into queryLDAPSecurityManager(IEspContext &context, bool required) will eliminate even more, but again that's for another JIRA if you decide you want to do it.

Reply:
The JIRA is created: https://track.hpccsystems.com/browse/HPCC-28570.